server/meshmodel/helper.go

```diff
@@ -0,0 +1,135 @@
+package meshmodel
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/layer5io/meshery/server/helpers/utils"
+	"github.com/layer5io/meshery/server/models"
+	"github.com/layer5io/meshkit/models/meshmodel"
+	"github.com/layer5io/meshkit/models/meshmodel/core/v1alpha1"
+	"github.com/pkg/errors"
+)
+
+const ArtifactHubComponentsHandler = "kubernetes" //The components generated in output directory will be handled by kubernetes
+
+type ComponentHelper struct {
+	handlerConfig    *models.HandlerConfig
+	regManager       *meshmodel.RegistryManager
+	componentChan    chan v1alpha1.ComponentDefinition
+	relationshipChan chan v1alpha1.RelationshipDefinition
+	doneSignal       chan bool
+	errorChan        chan error
+}
+
+func (ch *ComponentHelper) Init(hc *models.HandlerConfig, rm *meshmodel.RegistryManager) {
+	ch.handlerConfig = hc
```

Comment:
@Azanul No need to pass the handlerconfig here. Execute the SeedComponents inside an anonymous func go routine and once it returns, Call publish. No need to use doneChannel. The original code was already abusing go channels. I should have refactored before

Reply:
I think it's better encapsulated this way

Reply:
It is not the responsibility of this function to publish anything. Passing a handlerconfig is pointless. Main should handle that. It's unnecessary complexity. This function should do one job and that job properly.

Reply:
There is no further synchronization routine needed after registration ends. Hence we dont need channels. A simple rule of thumb is to ask "Do we need synchronisation here". What you are trying to synchronise is effectively same as returning at the end a boolean value. Therefore channels are being abused here. Normal returns take place on stack,  channels require heap. Unnecessary.

Reply:
How do you suggest closing `watchComponents` ?

Reply:
use context (create one and pass on watchComponents. Wait on ctx.Done() and call done() defer done() in SeedComponents)

Reply:
Whenever the question comes "how to make sure X exits when Y exits?". The first thing that should pop in the head is passing context and not using state variables. (Unless it is a complicated use case with multiple states interacting with each other.)

Reply:
context it is then